File: src/examples/r/vector-ops.R
export default {
  id: 'vector-ops-r',
  language: 'r',
  title: 'R Vector Operations',
  description: 'Vectorized operations, filtering, and statistics in R',
  difficulty: 'beginner',
  category: 'basics',
  tags: ['vectors', 'statistics', 'r'],
  code: `# Vector operations in R
values <- c(12, 27, 35, 8, 42, 19, 55, 31)

cat("Original:", values, "\n")

# Vectorized operations
squared <- values^2
cat("Squared:", squared, "\n")

# Filtering
above_mean <- values[values > mean(values)]
cat("Above mean:", above_mean, "\n")

# Statistics
cat(sprintf("Mean: %.2f\n", mean(values)))
cat(sprintf("Median: %.2f\n", median(values)))
cat(sprintf("SD: %.2f\n", sd(values)))
cat(sprintf("Sum: %.0f\n", sum(values)))

# Apply family
tripled <- sapply(values, function(x) x * 3)
cat("Tripled:", tripled, "\n")`
};

File: src/examples/rust/vector-ops.rs
/**
 * Vector Operations - Rust
 *
 * Common Rust Vec operations: push, pop, iter, map, filter.
 */
export default {
  id: 'vector-ops-rust',
  language: 'rust',
  title: 'Vector Operations — Rust',
  description: 'Rust Vec operations: push, map, filter, iter',
  difficulty: 'intermediate',
  category: 'basics',
  tags: ['rust', 'vectors', 'collections', 'iterators'],
  code: `fn main() {
    // Create and populate a Vec
    let mut numbers: Vec<i32> = Vec::new();
    for i in 1..=5 {
        numbers.push(i * 10);
    }
    println!("Original: {:?}", numbers);

    // Map: double each element
    let doubled: Vec<i32> = numbers.iter().map(|&x| x * 2).collect();
    println!("Doubled: {:?}", doubled);

    // Filter: keep only values > 30
    let filtered: Vec<i32> = numbers.iter()
        .filter(|&&x| x > 30)
        .copied()
        .collect();
    println!("Filtered (>30): {:?}", filtered);

    // Fold (reduce): sum
    let sum: i32 = numbers.iter().sum();
    println!("Sum: {}", sum);

    // Max and min
    let max = numbers.iter().max().unwrap();
    let min = numbers.iter().min().unwrap();
    println!("Max: {}, Min: {}", max, min);

    // Sort descending
    let mut desc = numbers.clone();
    desc.sort_by(|a, b| b.cmp(a));
    println!("Sorted descending: {:?}", desc);
}`,
};
